packages/framework/attributor/src/runtimeAttributorDataStoreChannel.ts

```diff
@@ -0,0 +1,286 @@
+ */
+
+import { TypedEventEmitter } from "@fluid-internal/client-utils";
+import { AttachState, IDeltaManager } from "@fluidframework/container-definitions/internal";
+import {
+	FluidObject,
+	IFluidHandle,
+	IRequest,
+	IResponse,
+} from "@fluidframework/core-interfaces";
+import type { IFluidHandleInternal } from "@fluidframework/core-interfaces/internal";
+import { assert, Deferred, unreachableCase } from "@fluidframework/core-utils/internal";
+import { FluidObjectHandle } from "@fluidframework/datastore/internal";
+import { IFluidDataStoreRuntimeEvents } from "@fluidframework/datastore-definitions/internal";
+import {
+	IDocumentMessage,
+	type ISnapshotTree,
+	ISequencedDocumentMessage,
+	IQuorumClients,
+} from "@fluidframework/driver-definitions/internal";
+import {
+	IGarbageCollectionData,
+	IFluidDataStoreChannel,
+	IFluidDataStoreContext,
+	IInboundSignalMessage,
+	VisibilityState,
+	type ISummaryTreeWithStats,
+	type ITelemetryContext,
+} from "@fluidframework/runtime-definitions/internal";
+import {
+	ITelemetryLoggerExt,
+	MonitoringContext,
+	raiseConnectedEvent,
+	createChildMonitoringContext,
+} from "@fluidframework/telemetry-utils/internal";
+
+import {
+	type IProvideRuntimeAttributor,
+	type IRuntimeAttributor,
+} from "./attributorContracts.js";
+import { RuntimeAttributor } from "./runtimeAttributor.js";
+
+/**
+ * Data store channel for the runtime attributor. This channel is responsible for storing and managing the
+ */
+export class RuntimeAttributorDataStoreChannel
+	extends TypedEventEmitter<IFluidDataStoreRuntimeEvents>
+	implements IFluidDataStoreChannel, IProvideRuntimeAttributor
+{
+	public constructor(
+		public readonly dataStoreContext: IFluidDataStoreContext,
+		existing: boolean,
+	) {
+		super();
+		this.runtimeAttributor = new RuntimeAttributor();
+		this.mc = createChildMonitoringContext({
+			logger: dataStoreContext.baseLogger,
+			namespace: "Attributor",
+		});
+		this._attachState = dataStoreContext.attachState;
+		if (existing) {
+			this.visibilityState =
+				dataStoreContext.attachState === AttachState.Detached
+					? VisibilityState.LocallyVisible
+					: VisibilityState.GloballyVisible;
+		} else {
+			this.visibilityState = VisibilityState.NotVisible;
+		}
+		// If it's existing we know it has been attached.
+		if (existing) {
+			this.deferredAttached.resolve();
+		}
+		this.entryPoint = new FluidObjectHandle<FluidObject>(
+			this.runtimeAttributor,
+			"",
+			dataStoreContext.IFluidHandleContext,
+		);
+	}
+
+	public get IFluidDataStoreChannel(): IFluidDataStoreChannel {
+		return this;
+	}
+
+	public get IRuntimeAttributor(): IRuntimeAttributor {
+		return this.runtimeAttributor;
+	}
+
+	private _disposed = false;
+	public get disposed(): boolean {
+		return this._disposed;
+	}
+
+	public dispose(): void {
+		this._disposed = true;
+	}
+
+	private readonly runtimeAttributor: RuntimeAttributor;
+	public isEnabled = true;
+	public _attachState: AttachState;
```

Comment:
nit: public with `_` prefix - is that intentional?

Reply:
fixed.